frontend/build.go

```diff
@@ -88,75 +67,29 @@ func fillPlatformArgs(prefix string, args map[string]string, platform ocispecs.P
 	}
 }
 
-// Build is the main entrypoint for the dalec frontend.
-func Build(ctx context.Context, client gwclient.Client) (*gwclient.Result, error) {
-	if !SupportsDiffMerge(client) {
-		dalec.DisableDiffMerge(true)
-	}
-
-	bc, err := dockerui.NewClient(client)
-	if err != nil {
-		return nil, fmt.Errorf("could not create build client: %w", err)
-	}
-
-	spec, err := loadSpec(ctx, bc)
-	if err != nil {
-		return nil, err
-	}
-
-	if err := registerSpecHandlers(ctx, spec, client); err != nil {
-		return nil, err
-	}
-
-	res, handled, err := bc.HandleSubrequest(ctx, makeRequestHandler(bc.Target))
-	if err != nil || handled {
-		return res, err
-	}
+type PlatformBuildFunc func(ctx context.Context, client gwclient.Client, platform *ocispecs.Platform, spec *dalec.Spec, targetKey string) (gwclient.Reference, *dalec.DockerImageSpec, error)
```

Comment:
For some reason the name `targetKey` has been breaking my brain, maybe because I associate key so heavily with caching and/or table lookup. I wonder if we could change this to `targetRoute`, since this is so conceptually similar to an HTTP router?

Reply:
In this case "targetKey" is litterally referencing the map key in the `Targets` field on the spec.

Reply:
Yeah, I think I picked a poor place to make this comment, apologies -- I guess it's not really relevant in this particular spot.  I meant to refer to the use of `targetKey` as the name for routes in the `BuildMux`.  Would you be open to renaming this to something that reflects its treatment as a `route` or `path` in the `BuildMux` specifically?

Reply:
I changed this in the `Add` function, but everywhere else it seems like the right wording.